server/memphis_handlers_monitoring.go

```diff
@@ -581,14 +581,33 @@ cleanup:
 			logSource, logType = splittedSubj[1], splittedSubj[2]
 		}
 
-		data := string(msg.Data)
-		resMsgs = append(resMsgs, models.Log{
-			MessageSeq: int(msg.Sequence),
-			Type:       logType,
-			Data:       data,
-			Source:     logSource,
-			TimeSent:   msg.Time,
-		})
+		filterSubjectType := strings.Split(cc.FilterSubject, ".")
+		var filterSub string
+		if len(strings.Split(cc.FilterSubject, ".")) == 2 {
+			filterSub = filterSubjectType[1]
+		} else if len(strings.Split(cc.FilterSubject, ".")) == 3 {
+			filterSub = filterSubjectType[2]
+		}
+
+		if filterSubject != "" && filterSub == logType {
+			data := string(msg.Data)
+			resMsgs = append(resMsgs, models.Log{
+				MessageSeq: int(msg.Sequence),
+				Type:       logType,
+				Data:       data,
+				Source:     logSource,
+				TimeSent:   msg.Time,
+			})
+		} else {
+			data := string(msg.Data)
+			resMsgs = append(resMsgs, models.Log{
+				MessageSeq: int(msg.Sequence),
+				Type:       logType,
+				Data:       data,
+				Source:     logSource,
+				TimeSent:   msg.Time,
+			})
+		}
```

Comment:
this `if` looks like it does the same on both cases?!

Reply:
which make me think... why do we need `filterSub` at all? the consumer should consume only logs by that filter in the first place